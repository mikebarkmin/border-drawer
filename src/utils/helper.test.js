import {
  calcSimilarity,
  calcArea,
  getBiggestPolygon,
  xyToLngLat,
  lngLatToXy
} from './helper';
import * as Turf from '@turf/turf';

import de from '../borders/de.json';

test('xyToLngLat', () => {
  const dimensions = {
    width: 800,
    height: 800
  };
  const point = [0, 0];
  expect(xyToLngLat(dimensions, point, de)).toMatchSnapshot();
});

test('getBiggestPolygon', () => {
  const points = [[0, 0], [800, 0], [800, 800], [0, 800], [0, 0]];
  const dimensions = {
    width: 800,
    height: 800
  };
  const coordinates = points.map(point => xyToLngLat(dimensions, point, de));

  const userPolygon = Turf.polygon([coordinates], {});
  const biggestPolygon = getBiggestPolygon(userPolygon);
  expect(biggestPolygon).toMatchSnapshot();
});

test('calcArea', () => {
  const points = [[0, 0], [800, 0], [800, 800], [0, 800], [0, 0]];
  const dimensions = {
    width: 800,
    height: 800
  };
  const coordinates = points.map(point => xyToLngLat(dimensions, point, de));

  const userPolygon = Turf.polygon([coordinates], {});
  const biggestPolygon = getBiggestPolygon(userPolygon);
  const area = calcArea(biggestPolygon, de);
  expect(area).toMatchSnapshot();
  const area2 = calcArea(de, biggestPolygon);
  expect(area2).toBe(0);
});

test('calcSimilarity', () => {
  let points = [[0, 0], [800, 0], [800, 800], [0, 800], [0, 0]];
  const dimensions = {
    width: 800,
    height: 800
  };
  let similarity = calcSimilarity(points, de, dimensions);
  expect(similarity).toMatchSnapshot();

  points = de.geometry.coordinates[0].map(coordinate =>
    lngLatToXy(dimensions, coordinate, de)
  );
  similarity = calcSimilarity(points, de, dimensions);
  expect(similarity).toBe(100);
});

test('lngLatToXy', () => {
  const lngLat = [5.988658074577813, 47.30248769793916];
  const dimensions = {
    width: 800,
    height: 800
  };
  const xy = lngLatToXy(dimensions, lngLat, de);
  expect(xy).toMatchSnapshot();
});
