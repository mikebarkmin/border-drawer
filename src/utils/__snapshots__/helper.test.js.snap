// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calcArea 1`] = `537021342724.8637`;

exports[`calcSimilarity 1`] = `-51`;

exports[`getBiggestPolygon 1`] = `
Object {
  "geometry": Object {
    "coordinates": Array [
      Array [
        Array [
          3.6725370367922907,
          55.29047320526871,
        ],
        Array [
          17.20158405695775,
          55.29047320526871,
        ],
        Array [
          17.20158405695775,
          46.81789095575502,
        ],
        Array [
          3.6725370367922907,
          46.81789095575502,
        ],
        Array [
          3.6725370367922907,
          55.29047320526871,
        ],
      ],
    ],
    "type": "Polygon",
  },
  "properties": Object {},
  "type": "Feature",
}
`;

exports[`lngLatToXy 1`] = `
Array [
  136.95693624736606,
  757.9356691862426,
]
`;

exports[`xyToLngLat 1`] = `
Array [
  3.6725370367922907,
  55.29047320526871,
]
`;
